### ISSUE TITLE
Post-train Quantization Element-wise addition function can't use

### ISSUE THREAD

**Issue by georgetseng1128** [NervanaSystems/distiller#302]
Hi, I try Post-train Quantization Element-wise addition function.

Command:

> python compress_classifier.py -a resnet18 --pretrained /dataset/imagenet2012 --evaluate --quantize-eval

but, RangeLinearQuantEltwiseAddWrapper not be used.

warning log:

> WARNING: module.layer1.0.add - quantization of EltwiseAdd without stats not supported. Keeping the original FP32 module

How do can use function RangeLinearQuantEltwiseAddWrapper?

**Comment by levzlotnik** [comment]
Hi @georgetseng1128 ,

Before quantizing the model with non-parametric layers (like `nn.ReLu` or `EltwiseAdd`) you should first collect the activation stats for this layer. Some layers support **dynamic quantization** (e.g. `nn.Linear` and `nn.Conv2d`), but these don't.

Collecting stats for the layer (or for the entire model) is as easy as adding `--activation-stats` argument to the `compress_classifier.py` command line.  
NOTE - this will slow down the runtime of the script significantly, so I suggest running it a single time, without the `--quantize-eval` argument.
A yaml file will be generated with all the activation stats for all the layers, and then this file can be used to feed the activation for the PostTrainLinearQuantizer, suppose it's called `activation_stats.yaml`.  
Use this file by adding an argument `--qe-stats-file activation_stats.yaml`, and you're good to go.

Let us know if you need more help :)
Lev

**Comment by guyjacob** [comment]
Actually, to collect activation stats **specifically for quantization** you need to use `--qe-calibration` in the command line, not `--activation-stats`. I realize this is a bit confusing, but that's they way it is at the moment :).
Also note that `--qe-calibration` expects a numeric value between [0 .. 1] which specifies how much of the test dataset you want to use for stats collection. For example - `--qe-calibration 0.1` will use 10% of the test set. That's usually a safe value to use.

You can see an example of a stats file with a bit more details here:
https://github.com/NervanaSystems/distiller/blob/master/examples/quantization/post_train_quant/stats/resnet18_quant_stats.yaml
It's actually for ResNet-18, so you can just use it as is :)

**Comment by georgetseng1128** [comment]
Thank you all!
I have successfully executed, use `--qe-calibration` and `--qe-stats-file`.

**Comment by georgetseng1128** [comment]
Thank you all!
I have successfully executed.
Use `--qe-calibration` and `--qe-stats-file`.
